package/MDAnalysis/lib/formats/xdrlib.pyx

```diff
@@ -259,10 +262,64 @@ cdef class _XDRFile:
         self.reached_eof = False
         ok = xdr_seek(self.xfp, offset, SEEK_SET)
         if ok != 0:
-            raise RuntimeError("XDR seek failed, error-message={}".format(
-                error_message[ok]))
+            # errno is the direct system error, not from the XDR library.
+            # errno=22 can be cryptic, since it means a wrong 'whence'
+            #  parameter but it will also be issued when the resulting file
+            #  offset is negative, or if the offset overflows the filesystem
+            #  limit (ext3 is 16TB, for instance).
+            raise RuntimeError("XDR seek failed with system errno={}".format(ok))
         self.current_frame = frame
 
+    def _bytes_seek(self, offset, whence="SEEK_SET"):
+        """Low-level access to the stream repositioning xdr_seek call.
+
+        Beware that this function will not update :attr:`current_frame`,
+        even on a successful seek, and might render the :class:`Reader`
+        unstable.
+
+        Parameters
+        ----------
+        offset : int
+            move the stream *offset* bytes relative to *whence*
+            (can be negative).
+
+        whence : str, optional
+            must be one of:
+             "SEEK_SET": *offset* is applied relative to the beginning
+                         of the file.
+             "SEEK_CUR": *offset* is applied relative to the current
+                         position in the stream.
+             "SEEK_END": *offset* is applied relative to the end of
+                         the file (a positive *offset* will seek beyond the
+                         end of the file, without an error)
+
+        Raises
+        ------
+        ValueError
+            If *whence* is not one of the expected strings.
+
+        RuntimeError
+            If the seek fails (the low-level system error is reported).
+        """
+        cdef int whn = SEEK_SET
+        cdef int ok
+        cdef int64_t offst
+
+        if whence == "SEEK_CUR":
+            whn = SEEK_CUR
+        elif whence == "SEEK_END":
+            whn = SEEK_END
+        elif whence != "SEEK_SET":
+            raise ValueError("Parameter 'whence' must be one of 'SEEK_SET', "
+                             "'SEEK_CUR', or 'SEEK_END'.")
+        offst = offset
+        self.reached_eof = False
+        ok = xdr_seek(self.xfp, offst, whn)
+        if ok != 0:
+            # See the comments to seek() for hints on errno meaning.
+            raise RuntimeError("XDR seek failed with "
```

Comment:
Use IOError. It fits better   I'm also correcting that in my PR for other error codes